From 15391e9d34605e25f9e9ea8e9dd1b04152476108 Mon Sep 17 00:00:00 2001
From: Daniele Palmas <dnlplm@gmail.com>
Date: Fri, 28 Jun 2024 10:23:07 +0200
Subject: [PATCH 24/26] net: mhi: fix wrong error checking in mhi_net_newlink

Fix the variable checked for memory allocation in function
mhi_net_newlink.

Signed-off-by: Daniele Palmas <dnlplm@gmail.com>
---
 drivers/net/mhi_net.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/mhi_net.c b/drivers/net/mhi_net.c
index 062e4abbd623..fb14b52f3ed6 100644
--- a/drivers/net/mhi_net.c
+++ b/drivers/net/mhi_net.c
@@ -368,17 +368,17 @@ static int mhi_net_newlink(struct mhi_device *mhi_dev, struct net_device *ndev)
 	mhi_netdev->ndev = ndev;
 	mhi_netdev->mdev = mhi_dev;
 	mhi_netdev->skbagg_head = kmalloc(sizeof(struct sk_buff *) * mhi_dev->num_rx_chan, GFP_KERNEL);
-	if (err)
+	if (!mhi_netdev->skbagg_head)
 		goto out_dev_free;
 	mhi_netdev->skbagg_tail = kmalloc(sizeof(struct sk_buff *) * mhi_dev->num_rx_chan, GFP_KERNEL);
-	if (err)
+	if (!mhi_netdev->skbagg_tail)
 		goto out_err_rx_skbagg;
 
 	mhi_netdev->mru = mhi_dev->mhi_cntrl->mru;
 
 	mhi_netdev->queue = mhi_dev->queue;
 	mhi_netdev->rx_queue_sz = kmalloc(sizeof(u32) * mhi_dev->num_rx_chan, GFP_KERNEL);
-	if (err)
+	if (!mhi_netdev->rx_queue_sz)
 		goto out_err_rx_skbagg_tail;
 
 	INIT_DELAYED_WORK(&mhi_netdev->rx_refill, mhi_net_rx_refill_work);
-- 
2.46.0

